api/handler/v1beta1/job_spec.go

```diff
@@ -163,30 +163,57 @@ func (sv *JobSpecServiceServer) CreateJobSpecification(ctx context.Context, req
 		return nil, mapToGRPCErr(sv.l, err, "unable to get namespace")
 	}
 
-	jobSpec, err := FromJobProto(req.GetSpec(), sv.pluginRepo)
+	reqJobSpec := req.GetSpec()
+	jobSpec, err := FromJobProto(reqJobSpec, sv.pluginRepo)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
 	}
+	jobSpecs := []models.JobSpec{jobSpec}
 
-	// validate job spec
-	if err = sv.jobSvc.Check(ctx, namespaceSpec, []models.JobSpec{jobSpec}, logWriter); err != nil {
-		return nil, status.Errorf(codes.Internal, "spec validation failed\n%s", err.Error())
+	deployID, err := sv.jobSvc.CreateAndDeploy(ctx, namespaceSpec, jobSpecs, logWriter)
+
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "job %s is created but deployment scheduling failed for project %s, error:: %s", jobSpec.Name, req.GetProjectName(), err.Error())
 	}
+	runtimeDeployJobSpecificationCounter.Inc()
 
-	_, err = sv.jobSvc.Create(ctx, namespaceSpec, jobSpec)
+	return &pb.CreateJobSpecificationResponse{
+		Success: true,
+		Message: fmt.Sprintf("job %s is created and queued for deployment on project %s, with DeploymentId :: %s", jobSpec.Name, req.GetProjectName(), deployID.UUID().String()),
+	}, nil
+}
+
+func (sv *JobSpecServiceServer) AddJobSpecifications(ctx context.Context, req *pb.AddJobSpecificationsRequest) (*pb.AddJobSpecificationsResponse, error) {
+	namespaceSpec, err := sv.namespaceService.Get(ctx, req.GetProjectName(), req.GetNamespaceName())
+	logWriter := writer.NewLogWriter(sv.l)
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "%s: failed to save job %s", err.Error(), jobSpec.Name)
+		return nil, mapToGRPCErr(sv.l, err, "unable to get namespace")
 	}
 
-	// TODO: remove progressObserver injection here
-	if err := sv.jobSvc.Sync(ctx, namespaceSpec, sv.progressObserver); err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to sync jobs: \n%s", err.Error())
+	reqJobSpecs := req.GetSpecs()
+	var jobSpecs []models.JobSpec
+	for _, spec := range reqJobSpecs {
+		jobSpec, err := FromJobProto(spec, sv.pluginRepo)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "cannot deserialize job: \n%s", err.Error())
+		}
+		jobSpecs = append(jobSpecs, jobSpec)
+	}
+
+	jobNames := make([]string, len(jobSpecs))
+	for i, jobSpec := range jobSpecs {
+		jobNames[i] = jobSpec.Name
 	}
+	deployID, err := sv.jobSvc.CreateAndDeploy(ctx, namespaceSpec, jobSpecs, logWriter)
 
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "jobs %s is/are created but deployment scheduling failed for project %s, error:: %s", strings.Join(jobNames, ", "), req.GetProjectName(), err.Error())
```

Comment:
I believe it will be too verbose in the log to put all of the job names. How about we limit it? For example, let's say we have 20 jobs being added, then we can have:
```
jobs [jobA, jobB] and 18 others are created but deployment ...
```

Reply:
Sure we can do that, 
but i thought we will be using this end point for deploying a small (2, 3 maybe 5 ) number  of jobs only.

Reply:
we don't need to mention the job names if we don't have the idea of which jobs it failed. We can just log the error. Otherwise it will give a misinterpretation that creation of all jobs failed.

Reply:
actually right, fixing that